src/igvfd/types/file_set.py

```diff
@@ -118,6 +118,76 @@ class CuratedSet(FileSet):
     schema = load_schema('igvfd:schemas/curated_set.json')
     embedded_with_frame = FileSet.embedded_with_frame
 
+    @calculated_property(
+        schema={
+            'title': 'Assembly',
+            'description': 'The genome assembly to which coordinates relate (e.g., GRCh38).',
+            'type': 'array',
+            'uniqueItems': True,
+            'items': {
+                'type': 'string'
+            },
+            'notSubmittable': True,
+        }
+    )
+    def assembly(self, request, files):
+        assembly_values = set()
+        if files is not None:
+            for current_file_path in files:
+                file_object = request.embed(current_file_path, '@@object?skip_calculated=true')
+                if file_object.get('assembly'):
+                    assembly_values.add(file_object.get('assembly'))
+        return list(assembly_values)
+
+    @calculated_property(
+        schema={
+            'title': 'Transcriptome Annotation',
+            'description': 'The annotation and version of the reference resource.',
+            'type': 'array',
+            'uniqueItems': True,
+            'items': {
+                'type': 'string'
+            },
+            'notSubmittable': True,
+        }
+    )
+    def transcriptome_annotation(self, request, files):
+        annotation_values = set()
+        if files is not None:
+            for current_file_path in files:
+                file_object = request.embed(current_file_path, '@@object?skip_calculated=true')
+                if file_object.get('transcriptome_annotation'):
+                    annotation_values.add(file_object.get('transcriptome_annotation'))
+        return list(annotation_values)
+
+    @calculated_property(
+        schema={
+            'title': 'Summary',
+            'type': 'string',
+            'notSubmittable': True,
+        }
+    )
+    def summary(self, request, curated_set_type, files, taxa=None):
+        message_list = []
+        message_list.append(curated_set_type)
+        if taxa:
+            message_list.append(taxa)
+        if files is not None:
+            assembly_values = set()
+            annotation_values = set()
+            for current_file_path in files:
+                file_object = request.embed(current_file_path, '@@object?skip_calculated=true')
+                if file_object.get('assembly'):
+                    assembly_values.add(file_object.get('assembly'))
+                if file_object.get('transcriptome_annotation'):
+                    annotation_values.add(file_object.get('transcriptome_annotation'))
+            for assembly_value in assembly_values:
+                message_list.append(assembly_value)
+            for annotation_value in annotation_values:
+                message_list.append(annotation_value)
+        summary_message = ' '.join(message_list)
```

Comment:
I don't think the summary should be constructed form a list, that could be re-sorted and then the summary would be broken. 
instead construct pieces of the summary and assemble them in the end

Reply:
Done